src/contexts/Api/index.tsx

```diff
@@ -66,36 +71,14 @@ export const APIProvider = ({ children }: { children: React.ReactNode }) => {
     !!localStorage.getItem('isLightClient')
   );
 
-  // initial connection
   useEffect(() => {
-    if (!provider) {
-      const _network: NetworkName = localStorage.getItem(
-        'network'
-      ) as NetworkName;
-      connect(_network, isLightClient);
-    }
-  });
+    connect(network.name, isLightClient);
+  }, []);
 
-  // provider event handlers
-  useEffect(() => {
-    if (provider !== null) {
-      provider.on('connected', () => {
-        setConnectionStatus(ConnectionStatus.Connected);
-      });
-      provider.on('error', () => {
-        setConnectionStatus(ConnectionStatus.Disconnected);
-      });
-      connectedCallback(provider);
-    }
-  }, [provider]);
-
-  // connection callback
-  const connectedCallback = async (_provider: WsProvider | ScProvider) => {
+  const recreateApi = async (_provider: WsProvider | ScProvider) => {
     const _api = new ApiPromise({ provider: _provider });
     await _api.isReady;
 
-    localStorage.setItem('network', String(network.name));
```

Comment:
Moved in [here](https://github.com/Cardinal-Cryptography/aleph-zero-dashboard/pull/37/files#diff-de701a1090eab6a6ce998b10578ba8d16bd3d81a57a7caf3ea8d1d14141ab0a8R187) to set the local storage _cache_ asap - it's after all what the user had chosen should to be set (and thuse consumed upon refreshes) and we should respect maintain that selection regardless of whether the actual connection or further processing succeeds or not.